handlers/update_resource.go

```diff
@@ -83,3 +72,42 @@ func (d *updateResourceEntry) addToStream(id string) error {
 	}
 	return d.stream.SendMessage(string(message))
 }
+
+func (d *updateResourceEntry) compareAndUpdateResource(id string, newResource datautils.Resource, existingResource *datautils.Resource) error {
```

Comment:
This seems like something that should go in `datautils` or `db`, it's not a great match for the `handler` package.

Reply:
Hmm... I'm open to discussion on this for sure. I don't think datautils is right (at least with the current implementation) as it would require dataUtils to be aware of the database. Maybe db, but then should db have that kind of logic? It seems like the bulk of the logic really be in the handler.

Reply:
My thinking here is that handler should mainly deal with web-interaction stuff.  E.g. identifying the user, and starting the database update, logging and streaming stuff.  This `compareAndUpdateResource` function is not specific to web interactions and could be used outside the web context, (Say if we had a background worker), as it's dealing only with data level concerns.

Reply:
That makes sense. I need to rework this a bit per this mornings discussion of updating when the version changes anyway, so I'll move some of this logic into dataUtils and the actual send to update will happen from the handler.